src/controller/movieController.js

```diff
@@ -1,67 +1,59 @@
-const movieModel = require("../models/Movie");
-const moviesDB = require("../database/movies");
+const movies = require("../models/Movie");
 
-showAll = (req, res) => {
-  res.json(moviesDB);
-};
+async function showAll(req, res) {
+  const films = await movies.find();
+  res.json(films);
+}
 
-showMovie = (req, res) => {
-  const movieId = req.params.id;
-  const movie = moviesDB.find(movie => movie.id == movieId);
+async function showMovie(req, res) {
+  const movieTitle = req.params.title;
+  const movie = await movies.findOne({ title: movieTitle });
   res.json(movie);
-};
-
-addMovie = (req, res) => {
-  const movie = new movieModel(
-    moviesDB.length,
-    req.body.title,
-    req.body.director,
-    req.body.genre,
-    req.body.releaseDate,
-    req.body.rating,
-    req.body.likes
-  );
-  moviesDB.push(movie);
-  res.send("Se ha añadido correctamente");
-};
-
-deleteMovie = (req, res) => {
-  const movieId = req.params.id;
-  const movie = moviesDB.find(movie => movie.id == movieId);
-  const movieToRemove = moviesDB.indexOf(movie);
-  moviesDB.splice(movieToRemove, 1);
-  res.send("Se ha eliminado correctamente");
-};
-
-updateMovie = (req, res) => {
-  const movieIdToUpdate = req.params.id;
-  const updatedMovie = req.body;
-  const movie = moviesDB.find(movie => movie.id == movieIdToUpdate);
+}
 
-  movie.title = updatedMovie.title || movie.title;
-  movie.director = updatedMovie.director || movie.director;
-  movie.genre = updatedMovie.genre || movie.genre;
-  movie.releaseDate = updatedMovie.releaseDate || movie.releaseDate;
-  movie.rating = updatedMovie.rating || movie.rating;
+async function addMovie(req, res) {
+  const movieToCreate = new movies(req.body);
+  const real = await movies.find({ title: movieToCreate.title });
+  if (real == null) {
```

Comment:
Yo suelo hacer un:
```suggestion
  if (real) {
     res.send('Ya existe');
     return;
  }
```

Y ya luego hago la lógica en caso de que todo vaya bien